### PR TITLE
Copy Prometheus scaler example to Keda repo

### LINKED ISSUE #334: Move prometheus scaler example to keda-manager repository
**Description**

Please move the [example](https://github.com/kyma-project/examples/tree/main/scale-to-zero-with-keda) to this repo examples.
Change the references in the [user documentation](https://github.com/kyma-project/keda-manager/blob/main/docs/user/04-20-demo-applications.md#keda-prometheus-scaler-example)

**Area**
 - Keda

**Reasons**

Kyma examples repository will be archived. We should move relevant examples to module repo.

**Assignees**

@kyma-project/technical-writers

**Attachments**

### PR DESCRIPTION
**Pull request by grego952**
**Description**

Changes proposed in this pull request:

- Copy Prometheus scaler example to the Keda repo

**Related issue(s)**
Fixes #334

### CODE REVIEW

**Review by anoipm — DISMISSED** [review]

File: docs/user/examples/scale-to-zero-with-keda/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+At first the worker Function is scaled down.
+When listing HPA for the Function, you will see that the current replica count is zero.
+ ```bash
+kubectl get hpa
+NAME                               REFERENCE                     TARGETS             MINPODS   MAXPODS   REPLICAS   AGE
+keda-hpa-worker-fn-scaled-object   Function/scalable-worker-fn   <unknown>/2 (avg)   1         5         0          27h
+
+ ```
+ Also, when listing Pods by Function name label, you will see only the build job's Pod. No runtime Pod is up.
```
Comment (NHingerl) [inline docs/user/examples/scale-to-zero-with-keda/README.md:51]:
```suggestion
2. List Pods by Function name label and check that you see only the build job's Pod. No runtime Pod is up.
```

File: docs/user/examples/scale-to-zero-with-keda/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ ```bash
+kubectl get pods -l serverless.kyma-project.io/function-name=scalable-worker-fn -w
+NAME                                   READY   STATUS      RESTARTS   AGE
+scalable-worker-fn-build-7s4rf-wjhvt   0/1     Completed   0          2m16s
+ ```
+
+Once you generate a load (even a single request), the non-zero request rate targeting the worker Function triggers scaling up of the worker Function's runtime Pods.
```
Comment (NHingerl) [inline docs/user/examples/scale-to-zero-with-keda/README.md:58]:
```suggestion
3. Generate a load (even a single request) and check that the non-zero request rate targeting the worker Function triggers scaling up of the worker Function's runtime Pods.
```

File: docs/user/examples/scale-to-zero-with-keda/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+ Call the HTTP proxy Function once:
+
+ ```bash
+ curl -H "Content-Type: application/json" -X POST -d '{"foo":"bar"}' https://incoming.{your_cluster_domain}
+ ```
+
+The message is pushed to the Kyma Eventing.
```
Comment (NHingerl) [inline docs/user/examples/scale-to-zero-with-keda/README.md:66]:
```suggestion
The message is pushed to Kyma Eventing.
```

File: docs/user/examples/scale-to-zero-with-keda/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+It takes time to scale up a Function from zero. But no message is lost as Eventing retries delivery of the message to the subscriber until a running worker Pod eventually consumes it.
+
+Observe worker Function scaling up from zero. You can notice it by watching Function Pods or HPA.
```
Comment (NHingerl) [inline docs/user/examples/scale-to-zero-with-keda/README.md:69]:
```suggestion
4. Observe worker Function scaling up from zero. You can notice it by watching Function Pods or HPA.
```

File: docs/user/examples/scale-to-zero-with-keda/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+```bash
+kubectl get hpa -w                                                        
+NAME                               REFERENCE                     TARGETS             MINPODS   MAXPODS   REPLICAS   AGE
+keda-hpa-worker-fn-scaled-object   Function/scalable-worker-fn   <unknown>/2 (avg)   1         5         0          27h
+keda-hpa-worker-fn-scaled-object   Function/scalable-worker-fn   0/2 (avg)           1         5         1          27h
+```
+
+Observe that the payload was eventually processed.
+
+Check the worker Function logs. `Processing ... {"foo":"bar"}` eventually appears:
```
Comment (NHingerl) [inline docs/user/examples/scale-to-zero-with-keda/README.md:91]:
```suggestion
5. Check the worker Function logs. `Processing ... {"foo":"bar"}` eventually appears:
```

File: docs/user/examples/scale-to-zero-with-keda/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+user code loaded in 0sec 0.783514ms
+Processing ...
+{"foo":"bar"}
+
+ ```
+ 
+ If the traffic stops, the worker Function is scaled down back to zero replicas (after a configurable cooldown period).
+ 
+ If you generate a much higher load, for example,> 2 req/sec - as configured in the threshold value of the scaledObject, you will observe scaling up to more replicas. One replica must be added for each additional 2req/sec measured.
```
Comment (NHingerl) [inline docs/user/examples/scale-to-zero-with-keda/README.md:106]:
```suggestion
6. If you generate a much higher load, for example,> 2 req/sec - as configured in the threshold value of the scaledObject, you will observe scaling up to more replicas. One replica must be added for each additional 2req/sec measured. 
```

File: docs/user/examples/scale-to-zero-with-keda/assets/scaling-scenario.drawio [hunk 7]
```diff
@@ -0,0 +1 @@
...UC65GfUZKtpztbmEuEHJOj83UwsNBYFQ+08htq8/TdpCyqO70cV4porcrZSVONptFcgmxIA7OAU6b5wW9ZhUJ7pKssfcQT6nfjx1xcbKxzcEav72Yfsryrb5UVLMjg2A9XrC8Ctbn8R1JqpkPkjgid6nEyO7rBUQmV0Gg8o8SNKwPYT7FpHBK8D9PzjAraCXJ5YMPKzdMktfubP6oG+GFccBqeDGcoAGconwdemllqV+KWAKsJpnhgvb7Np9YeidgwPXRKe4c5zlrbn99eLKnV3B4w8DylDwllC9HiOOUjcttCm+/ikH/klqe+h5ls+lVIKQdeNE9HqGOdF/FEUdg0FTmYeNqOo5j4Eamkcazw2RGJ6MnQ32s1yoVcLsZcUUNeLhoKytg3H1p2IhVKj4tkjTMJyGCTWLjqM3jkf0Cxi8jxmvNrdM+GwKsS7/UkeXS+mJAtFmAWxA8SyjBdOmxrepmVjM0L4dc6cQ0VtkPs7BNSbX8afGrxCP5sLvNJLIfA2zxuQoKPVyQhYNlGdSmkBfQ8X9L2jWFuB88xd63xnB/kwxT9z7ukezFCBIv+H0sKS4NBhf4UkBaB8/gGYV0j4o0OuU0mNrULXRFEbdWRJYwHijscxK0AzJ+6XO8n533sz42Hh6eIOIee0O2A+tjeW4dNxZpRAsR5A9zcsaWIAeK+qvMKLJ7FPw6jLHEOp9OfkZethiBry/ewuVE77ESfB/dBy/3cco6zqvNr0omklgL/HIJZRH33ftOvvsWy8xlS+zbddO6tPs1omLJf4fcI68kKFwd9sVsjTt5jckjs3993knD2JxTkdhajmCJDH+GDdr0kyMMX0Bb7VZqBnmVKfKwOclwtPYOhlMVSM+df/Yvqs1KV7JrAlZLO7aD51hmvm+mlgR5t/u3DR/F2fsmXW9tP6l3HGULvlHo73MhDcSCvN+vTL3JCnTJWBR/mLeTICgeii+OlhhnGJUHl/P9zJm64V1yf70omBHdyvvzyP8+K2xXdLULAykqhkuwpvjm3+b367jdAyfWgfgCIFnA7o47ubd16+WNzDvUUskYFJM4xg+o68kmN81G5MdTi/9ILqi6AS78+hf41ODkDPj7f3JuiACX9v7rgS+U+jfk/z47hBD/9p/Hnf/liBz2P7gvEvj3CVzI/l9HGcHdL8ZUgHOjwr8D</diagram></mxfile>
```
Comment (NHingerl) [inline docs/user/examples/scale-to-zero-with-keda/assets/scaling-scenario.drawio:1]: this doesn't match our diagram guidelines :/

File: docs/user/04-20-demo-applications.md [hunk 8]
```diff
@@ -40,13 +40,13 @@ keda-demo   keda-hpa-orders-service-cpu-scaler   Deployment/orders-service   80%
...
 
 It demonstrates an event-driven approach that allows you to decouple functional parts of an application and apply consumption-based scaling.
 
 It uses: 
- - Functions to deploy workloads directly from a Git repository ([Kyma Serverless](https://kyma-project.io/docs/kyma/latest/01-overview/serverless/)),
- - In-cluster Eventing to enable event-driven communication ([Kyma Eventing](https://kyma-project.io/docs/kyma/latest/01-overview/eventing/)), 
+ - Functions to deploy workloads directly from a Git repository ([Kyma Serverless](https://kyma-project.io/#/serverless-manager/user/README)),
+ - In-cluster Eventing to enable event-driven communication ([Kyma Eventing](https://kyma-project.io/#/eventing-manager/user/README)), 
  - Prometheus and Istio to deliver metrics essential for scaling decisions,
  - Keda to drive the scaling.
```
[the remark this answers is not in the document]
Comment (grego952) [reply docs/user/04-20-demo-applications.md:51]: I use `KEDA` when I write about the [KEDA autoscaler](https://keda.sh/) and `Keda` when I write about the module. Definitely, I need to change this occurrence to KEDA :)

File: docs/user/examples/scale-to-zero-with-keda/README.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+It realises the following scenario:
+
+![scenario](assets/scaling-scenario.png "Scenario")
+
+The proxy Function receives the incoming HTTP traffic, and with every request, it publishes the payload as an in-cluster event to a particular topic.
+
+The second Function (the actual worker) is subscribed to the topic and processes the incoming messages. Until there are no messages published for its subscribed topic, it remains scaled to zero - there are no actual worker Pods living in the runtime.
+
+Keda is used to scale the worker Function. [KEDA Prometheus scaler](https://keda.sh/docs/latest/scalers/prometheus/) is used to measure the load targeted for the worker Function and scale it accordingly (from 0 to 5 replicas).
```
[the remark this answers is not in the document]
Comment (grego952) [reply docs/user/examples/scale-to-zero-with-keda/README.md:19]: See my previous comment.

File: docs/user/examples/scale-to-zero-with-keda/assets/scaling-scenario.drawio (hunk 7 above)
Comment (grego952) [reply docs/user/examples/scale-to-zero-with-keda/assets/scaling-scenario.drawio:1]: I'll work on it.

**Review by NHingerl — APPROVED** [review]

**Pull request merged**